### ISSUE TITLE
[GDA] [SUBGRAPH] Calculation support for the total distribution from a pool

### ISSUE THREAD

**Issue by vmichalik** [superfluid-finance/protocol-monorepo#1762]
## Overview

**What?**

An example code, perhaps solidified in existing sdk-core, working in conjunction with GDA-enabled protocol subgraph to calculate total distribution from a GDA pool to a member of the pool.

**How?**

- [x]   Add a field in Subgraph entity that helps the calculation.
- [ ]   Create example code that does the calculation.
  - Consider to add the code to the existing SDK?

**Why?**

A common development task when using a GDA, is to be able to calculate how much token has distributed through a GDA Pool to a member of the pool. Due to the nature of how GDA works, it is not a simple view function that you can query on-chain directly. Instead, a combination of subgraph indexing and off-chain code can support the calculation of the total distribution from a GDA pool to a member of the pool.

**Deadline:**

Used bu AverageX.

**UI/UX Design:**
None.

## Acceptance criteria
 - [ ] ...
 - [ ] Satisfies entire acceptance criteria
 - [ ] Consider communication & visibility
 - [ ] Code review is approved by at least 2 reviewers
 - [ ] Tested by Elvijs/QA & discovered bugs resolved
 - [ ] Documentation updated
 - [ ] Assignee/opened by merges PR
 - [ ] Automated E2E tests for happy path
 - [ ] Code is merged
 - [ ] Observability of critical user path
 - [ ] Steps to access code change as a user

## Links/Resources
 * ...

## Readiness criteria
- [ ] Descriptive title
- [ ] What?
- [ ] Why?
- [ ] How?
- [ ] Acceptance criteria complying with Definition of Done
- [ ] Estimate/Story points
- [ ] Deadline: If not a part of an epic
- [ ] Priority: Reflected as order on the board, walking the board from the top and right
- [ ] Assignee
- [ ] Labels
- [ ] Link to epic: If belonging to an epic
- [ ] A task is always a separate ticket/issue
- [ ] UI/UX Design (optional)
- [ ] Blockers/Prerequisites (optional): Use Zenhub blocked by/blocking feature

**Comment by hellwolf** [comment]
test

**Comment by kasparkallas** [comment]
Will investigate mapping `totalAmountReceivedUntilUpdatedAt` onto PoolMember to ease the calculation, like it was the case with IndexSubscriptions.

**Comment by kasparkallas** [comment]
> Will investigate mapping `totalAmountReceivedUntilUpdatedAt` onto PoolMember to ease the calculation, like it was the case with IndexSubscriptions.

Yep, doable with little effort.

**Comment by kasparkallas** [comment]
Removed the link because I'm not sure that PR should quite close this. But we added `totalAmountReceivedUntilUpdatedAt` to a PoolMember in that PR.

**Comment by kasparkallas** [comment]
The example code and SDK haven't been done. The field was added to subgraph though and being used in Console and AverageX.

**Comment by hellwolf** [comment]
What is AverageX?

**Comment by hellwolf** [comment]
@youssefea please back reference the code snippets here when you get it, so that we can close it.

**Comment by kasparkallas** [comment]
There's a bug that `poolMember.totalAmountReceivedUntilUpdatedAt` resets to 0 when the poolMember units go to zero. The `poolMember.totalAmountReceivedUntilUpdatedAt` shouldn't reduce, ever.

EDIT: actually maybe false alert

EDIT 2: actually not a false alert and the issue was fixed: https://github.com/superfluid-finance/protocol-monorepo/pull/1816
